src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -569,22 +568,28 @@ public unsafe static string Join(char separator, string[] value, int startIndex,
             // Defer argument validation to the internal function
             return JoinCore(&separator, 1, value, startIndex, count);
         }
-    
+
         // Joins an array of strings together as one string with a separator between each original string.
         //
-        public static string Join(string separator, params string[] value)
+        public unsafe static string Join(string separator, params string[] value)
         {
             if (value == null)
             {
                 throw new ArgumentNullException(nameof(value));
             }
-            return Join(separator, value, 0, value.Length);
+            if (string.IsNullOrEmpty(separator))
+                return string.Concat(value);
```

Comment:
:+1: This should help perf, since `Concat` has to do less bookkeeping than `Join`. Please use braces here, though.

```cs
if (string.IsNullOrEmpty(separator))
{
    return string.Concat(value);
}
```